### PR TITLE
HPCC-30220 ECL Watch fix authentication & ESP session lock issues

### PR DESCRIPTION
**Pull request by jeclrsg**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/eclwatch/stub.js [hunk 1]
```diff
@@ -35,13 +35,13 @@ define([
...
-                    window.location.replace(`/esp/files/index.html#/workunits/${params.Wuid}`);
+                    window.location.replace(`/#/workunits/${params.Wuid}`);
                     break;
                 case "GraphsWUWidget":
-                    window.location.replace(`/esp/files/index.html#/workunits/${params.Wuid}/metrics`);
+                    window.location.replace(`/#/workunits/${params.Wuid}/metrics`);
                     break;
                 default:
-                    window.location.replace("/esp/files/index.html");
+                    window.location.replace("/");
```
Comment (GordonSmith) [inline esp/src/eclwatch/stub.js:44]: Sanity checks:
* Did you test this works when switching between V9 and dojo pages?
* Does this code match what we have in the 9.2.x branch now?

**Review by GordonSmith — CHANGES_REQUESTED** [review]
@jeclrsg we should discuss, I don't think we should be back porting anything to the React V9 codebase, given it was a tech preview in 8.10.
If there are fixes needed for the Dojo pages, that is fine?

**Comment by GordonSmith** [comment]
@jeclrsg bump

File: esp/src/eclwatch/stub.js (hunk 1 above)
Comment (jeclrsg) [reply esp/src/eclwatch/stub.js:44]: I did test that this works as expected when switching versions. But no, it doesn't match what's currently in 9.2. I thought the other PR (#17843) targeting 8.12.x should be merged up into 9.0 and so on. Also, regarding back porting, I believe Stu had pointed out in an ECL Watch meeting that the folks who'd encountered this had only fairly recently moved to 8.10 and they could be there for a while yet.

**Comment by FernSe01** [comment]
Any plans to have the changes addressed for future merge on 8.10 release? This will benefit Quod client in Brazil.

**Pull request merged**